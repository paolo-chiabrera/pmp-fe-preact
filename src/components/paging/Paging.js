import { h } from 'preact';

export default function Paging({ currentPage, setPage }) {
	return (
		<div class="paging">
            <p>Current page: {currentPage}</p>
            <button onClick={() => setPage(currentPage-1)}>Prev</button>
            <button onClick={() => setPage(currentPage+1)}>Next</button>
        </div>
	);
}