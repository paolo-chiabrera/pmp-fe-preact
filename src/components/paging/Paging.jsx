import { h } from 'preact';
import { Link } from 'react-router-dom';

import { nextPageNumber, prevPageNumber } from '../../helpers';

export default function Paging({ pageNumber = 0 }) {
  return (
    <div class="paging over-footer">
      <Link to={`/pages/${prevPageNumber(pageNumber)}`}>
        <div class="icon-arrow-left" />
      </Link>
      <div class="current">{pageNumber}</div>
      <Link to={`/pages/${nextPageNumber(pageNumber)}`}>
        <div class="icon-arrow-right" />
      </Link>
    </div>
  );
}
