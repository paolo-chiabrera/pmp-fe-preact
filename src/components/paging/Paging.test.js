import { h } from 'preact';

import render from 'preact-render-to-string';

import { StaticRouter } from 'react-router';

import Paging from './Paging.jsx';

describe('Given a Paging component', () => {
  describe('When the page is 0', () => {
    it('should render correctly', () => {
      const tree = render(
        <StaticRouter location="/pages/0">
          <Paging pageNumber={0} />
        </StaticRouter>
      );

      expect(tree).toMatchSnapshot();
    });
  });

  describe('When the page is 1', () => {
    it('should render correctly', () => {
      const tree = render(
        <StaticRouter location="/pages/1">
          <Paging pageNumber={1} />
        </StaticRouter>
      );

      expect(tree).toMatchSnapshot();
    });
  });
});
