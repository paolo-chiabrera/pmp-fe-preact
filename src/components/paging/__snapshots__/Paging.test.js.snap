// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Paging component When the page is 0 should render correctly 1`] = `"[object Object]"`;

exports[`Given a Paging component When the page is 1 should render correctly 1`] = `"[object Object]"`;
